### PR TITLE
Adding BaseNormalization Class

### PR DESCRIPTION
**Pull request by SalmanMesia**
Added utils/normalization

This includes BaseNormalization class that can be used to create identity, indicator, extension-definition, malware, and infrastructure Stix Domain Objects

Consumer has the ability to apply logic and decide which SDOs to include, and when to include them based on the logic defined in results_translator.py when producing the final stix bundle

Each SDO also includes SDO validation, by passing the '--stix-validator' flag

### CODE REVIEW

**Comment by BrijJhala** [comment]
@mdazam1942 thank you for your time to review our utils changes. It would not impact on existing connector codebase. Its for normalization of third party threat feed report.

**Comment by mdazam1942** [comment]
> @mdazam1942 thank you for your time to review our utils changes. It would not impact on existing connector codebase. Its for normalization of third party threat feed report.

will start the review soon.

**Comment by delliott90** [comment]
Where is this class being used? It would be good to have some documentation around this.

**Comment by SalmanMesia** [comment]
> Where is this class being used? It would be good to have some documentation around this.

added a README file on how to use it

File: stix_shifter_modules/reversinglabs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+uuid
```
Comment (mdazam1942) [inline stix_shifter_modules/reversinglabs/requirements.txt:1]: you should set the specific version of the library.

File: stix_shifter_modules/reversinglabs/stix_translation/query_constructor.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                        "DestinationIpV4": ["ipv4", "ipv4_cidr"],
+                        "DestinationIpV6": ["ipv6"]}
+
+logger = logging.getLogger(__name__)
+data = ""
+dataType = ""
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
```
Comment (delliott90) [inline stix_shifter_modules/reversinglabs/stix_translation/query_constructor.py:22]: This shouldn't be needed as they are defined in the `operators.json` and brought in as part of the `BaseQueryTranslator` class. If you look at other connectors, you will see `self.comparator_lookup = self.dmm.map_comparator()` in the `QueryStringPatternTranslator` initialization.

File: stix_shifter_modules/reversinglabs/stix_translation/results_translator.py [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+      return 'anomalous-activity'
+    elif value in ['malicious', 'MALICIOUS']:
+      return 'malicious-activity'
+    # if key/value pair is not included in the XFE report
+    elif value is None:
+      return
+    else:
+      return 'unknown'
+
+  def find_hash_type_by_length(self, value):
```
Comment (delliott90) [inline stix_shifter_modules/reversinglabs/stix_translation/results_translator.py:25]: This would be a good candidate to move to `stix_shifter_utils/utils/helpers.py`

Comment (SalmanMesia) [reply stix_shifter_modules/reversinglabs/stix_translation/results_translator.py:25]: We can look into it on a later PR

File: stix_shifter_utils/normalization/base_normalization_readme.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
+In this STIX Shifter Connector, instead of returning a STIX Observation, we are returning a STIX 2.1 Bundle.
```
Comment (delliott90) [inline stix_shifter_utils/normalization/base_normalization_readme.md:1]: I would reword this line. This readme is describing a particular type of stix-shifter connector that uses the base normalization class to return a bundle of SDOs. The difference is that the regular json-to-stix flow returns a bundle of STIX 2.1 observed-data objects, where connectors using this normalization class return other SDO.

File: stix_shifter_modules/reversinglabs/stix_translation/results_translator.py [hunk 5]
```diff
@@ -0,0 +1,352 @@
...
+            return {"threat_score": round(UNKNOWN_SCORE_MIN, 1)}
+          elif(indicator['indicator_types'][0] == 'anomalous-activity'):
+            return {"threat_score": round(SUSPICIOUS_SCORE_MIN, 1)}
+          else:
+            return {"threat_score": round(MALICIOUS_SCORE_MIN, 1)}
+
+
+  def translate_results(self, data_source, data):
+    """
+    Translates JSON data into STIX results based on a mapping file
```
Comment (delliott90) [inline stix_shifter_modules/reversinglabs/stix_translation/results_translator.py:283]: Is the to-stix mapping file even used? I notice that it's an empty object in the JSON.

Comment (SalmanMesia) [reply stix_shifter_modules/reversinglabs/stix_translation/results_translator.py:283]: Its not used, I can update the description

File: stix_shifter_modules/reversinglabs/stix_translation/query_constructor.py (hunk 2 above)
Comment (SalmanMesia) [reply stix_shifter_modules/reversinglabs/stix_translation/query_constructor.py]: Updated

File: stix_shifter_utils/normalization/base_normalization_readme.md (hunk 4 above)
Comment (SalmanMesia) [reply stix_shifter_utils/normalization/base_normalization_readme.md]: Reworded

File: stix_shifter_modules/reversinglabs/stix_translation/results_translator.py (hunk 5 above)
Comment (SalmanMesia) [reply stix_shifter_modules/reversinglabs/stix_translation/results_translator.py]: Updated description

**Review by mdazam1942 — APPROVED** [review]

**Review by delliott90 — APPROVED** [review]
